### PR TITLE
Fix Format fields with underscore in Loupe Adapter

### LINKED ISSUE #368: [Loupe] Object fields are not normalized with underscore in sort conditions
With the following schema:

```php
return new Index('guest_session', [
    'uuid' => new Field\IdentifierField('uuid'),
    'dates' => new Field\ObjectField('dates', [
        'start_timestamp' => new Field\IntegerField('start_timestamp', sortable: true),
        'end_timestamp' => new Field\IntegerField('end_timestamp', sortable: true),
    ]),
]);
```

`dates.start_timestamp` is normalized to `dates_start_timestamp` when the index is created (https://github.com/schranz-search/schranz-search/blob/0.3/packages/seal-loupe-adapter/src/LoupeHelper.php#L126).

When I apply a sort condition with the `SearchBuilder`, the field is not normalized with an underscore, which throw an exception.

```php
use Schranz\Search\SEAL\Search\Condition;

$result = $this->engine->createSearchBuilder()
    ->addIndex('guest_session')
    ->addSortBy('dates.start_timestamp', 'asc')
    ->getResult();
```
```
Loupe\Loupe\Exception\SortFormatException: Cannot sort by "dates.start_timestamp". It must be defined as sortable attribute.
```

### PR DESCRIPTION
**Pull request by ker0x**
Fix #368

### CODE REVIEW

File: packages/seal-loupe-adapter/src/LoupeHelper.php [hunk 1]
```diff
@@ -98,6 +98,11 @@ public function reset(): void
         $this->loupe = [];
     }
 
+    public function formatField(string $field): string
```
Comment (alexander-schranz) [inline packages/seal-loupe-adapter/src/LoupeHelper.php:104]: We should mark this method as `@internal`

```suggestion
    /**
     * @internal
     */
    public function formatField(string $field): string
```

File: integrations/spiral/src/Console/IndexCreateCommand.php [hunk 2]
```diff
@@ -29,10 +29,10 @@
 final class IndexCreateCommand extends Command
 {
     #[Option(name: 'engine', mode: InputOption::VALUE_REQUIRED, description: 'The name of the engine')]
-    private string|null $engineName = null;
+    private ?string $engineName = null;
```
Comment (alexander-schranz) [inline integrations/spiral/src/Console/IndexCreateCommand.php]: The code style was updated. Please rebase to 0.3

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Thx for the pull request.

Looks good, just some small change request to get this merged.

I need to check if we can maybe with the current test fixtures create a reproducable failing test case. But I think at current state the `memory` adapter is very minimal doesn't supported already this kind of nested properties.

**Comment by ker0x** [comment]
@alexander-schranz i'va rebase and apply your suggestions 😉

**Comment by alexander-schranz** [comment]
@ker0x thank you!

**Pull request merged**